assets/js/blocks/product-query/variations/product-query.tsx

```diff
@@ -19,6 +19,11 @@ import {
 
 const VARIATION_NAME = 'woocommerce/product-query';
 
+// This is a feature flag to enable the custom inherit Global Query implementation.
+// This is not intended to be a permanent feature flag, but rather a temporary.
+// https://github.com/woocommerce/woocommerce-blocks/pull/7382
+const isCustomInheritGlobalQueryImplementationEnabled = false;
```

Comment:
What do you think about using an experimental [flag](https://github.com/woocommerce/woocommerce-blocks/blob/2646666a8ebe1e4813e98087268048dfcd9c3445/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md?plain=1#L21-L23) here instead? I think we just need to limit the global query support to experimental/development build only. 🤔

Reply:
> I think we just need to limit the global query support to experimental/development build only.

At least for now, we want to show the option inherited query from the template from Gutenberg. The Product Query is a foundation for other blocks, too (for example, `Buy it again`), and having `two inherit query from template` options can be confusing for the developers that check the block for the first time.

Also, this is a super WIP code because there is still a discussion about how we should approach the problem: this is why I used this approach.

It is fine to change it for me, but I think it could be not very clear.

Reply:
> The Product Query is a foundation for other blocks, too (for example, Buy it again), and having two inherit query from template options can be confusing for the developers that check the block for the first time.

Make senses!

Can you explain how can we use this flag for development later? I mean what do we need to do to enable the flag? At glance, we have to temporarily change the flag in the source code, right?

Reply:
> Can you explain how can we use this flag for development later? I mean what do we need to do to enable the flag? At glance, we have to temporarily change the flag in the source code, right?

Yes! Just set `isCustomInheritGlobalQueryImplementationEnabled` and `is_custom_inherit_global_query_implementation_enabled` to true.

Reply:
Maybe adding a note here to update the PHP flag too? (and vice-versa)

Reply:
It makes sense! I updated the comments!